src/source-map.ts

```diff
@@ -0,0 +1,192 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as acorn from 'acorn';
+import * as dom5 from 'dom5';
+import * as parse5 from 'parse5';
+import {AnalysisContext} from 'polymer-analyzer/lib/core/analysis-context';
+import {RawSourceMap, SourceMapConsumer, SourceMapGenerator} from 'source-map';
+import * as urlLib from 'url';
+
+import * as matchers from './matchers';
+
+const inlineSourcemapPrefix =
+    '\n//# sourceMappingURL=data:application/json;charset=utf8;base64,';
+
+const sourceMappingUrlExpr = /\n\/\/# sourceMappingURL=(.*)\n?/;
+
+/**
+ * Creates an identity source map from JS script content. Can offset original
+ * line/column data for inline script elements.
+ */
+export function createJsIdentitySourcemap(
+    sourceUrl: string,
+    sourceContent: string,
+    lineOffset: number = 0,
+    firstLineCharOffset: number = 0) {
```

Comment:
You can emit ES5, or you can emit ES6. I think that's similar to what closure compiler supports? We're choosing ES6 and just limiting ourselves for now.